assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/vision/track_predict.py

```diff
@@ -0,0 +1,196 @@
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        print("Inside load_context()")
+
+        if self._task_type in ["video-multi-object-tracking"]:
+            """
+            Install mmtrack, mmcv and mmdet using mim, with pip installation is not working
+            1. for mmtrack, one of its dependency is mmcv 1.6.2, which will trigger cuda related issues.
+                to mitigate, we use no dependency install for mmtrack, and put other dependencies in pip requirement
+            2. for opencv, the default installed by mmcv is opencv-python. however, it's installing unwanted UI,
+                which causes problems for stability. thus we force reinstall opencv-python-headless.
+            3. for numpy, we are reinstalling numpy to older version to be compatible to opencv. more info:
+                https://stackoverflow.com/questions/20518632/importerror-numpy-core-multiarray-failed-to-import
+            """
+            subprocess.check_call([sys.executable, "-m", "mim", "install", "mmcv-full==1.7.1"])
+            subprocess.check_call([sys.executable, "-m", "mim", "install", "mmdet==2.28.2"])
+            subprocess.check_call([sys.executable, "-m", "mim", "install", "--no-deps", "mmtrack==0.14.0"])
+            subprocess.check_call([sys.executable, "-m", "pip", "install",
+                                   "opencv-python-headless==4.7.0.72", "--force-reinstall"])
+            subprocess.check_call([sys.executable, "-m", "pip", "install", "numpy==1.19.3", "--force-reinstall"])
+
+            # importing mmdet/mmcv after installing using mim
+            from mmtrack.apis import init_model, inference_mot
+            import mmcv
+            self._inference_detector = inference_mot
+            self.video_reader = mmcv.VideoReader
+
+            try:
+                model_config_path = context.artifacts[MMDetLiterals.CONFIG_PATH]
+                model_weights_path = context.artifacts[MMDetLiterals.WEIGHTS_PATH]
+
+                if not torch.cuda.is_available():
+                    raise RuntimeError("CUDA is not available. GPU needed for MOT inference!")
+                self._config = mmcv.Config.fromfile(model_config_path)
+                self._model = init_model(self._config, model_weights_path)
+
+                print("Model loaded successfully")
+            except Exception:
+                print("Failed to load the the model.")
+                raise
+
+        else:
+            raise ValueError(f"invalid task type {self._task_type}."
+                             f"Supported tasks: video-multi-object-tracking")
+
+    def predict(self, context: mlflow.pyfunc.PythonModelContext, input_data: pd.DataFrame) -> pd.DataFrame:
+        """Perform inference on the input data.
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        :param input_data: Input images for prediction
```

Comment:
The description for input_data needs to change. Also the type description needs to be related to "video" column

Reply:
changed :)